### PR TITLE
Fix a bug with how ScrollableGrid deals with padding.

### LINKED ISSUE #5522: Pesto: Topmost item remains invisible till it is scrolled to its image.
While scrolling down, the topmost item will only become visible when the item is scrolled far enough down till its image is on the visible scroll surface.

Video: https://www.youtube.com/watch?v=zHq_F25WLtE

### PR DESCRIPTION
**Pull request by mpcomplete**
It wasn't taking padding into account when deciding which children were
visible. I modeled the solution off of the way ScrollableList handles
this.

Fixes https://github.com/flutter/flutter/issues/5522

### CODE REVIEW

**Comment by abarth** [comment]
Looks good but needs a test.

File: packages/flutter/lib/src/widgets/scrollable_grid.dart [hunk 1]
```diff
@@ -126,16 +126,24 @@ class GridViewport extends VirtualViewportFromIterable {
...
-    this.startOffset,
+    this.scrollOffset,
     this.delegate,
     this.onExtentsChanged,
     this.children
   }) {
     assert(delegate != null);
   }
 
+  /// The [startOffset] without taking the [padding] into account.
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/scrollable_grid.dart]: I think you mean the delegate's padding.

**Comment by mpcomplete** [comment]
Added a test.

**Comment by HansMuller** [comment]
LGTM. Travis wants some lint love.

**Pull request merged**